crates/flox-rust-sdk/src/models/environment.rs

```diff
@@ -188,22 +208,27 @@ impl<S> Environment for PathEnvironment<S>
 where
     S: TransactionState,
 {
-    /// Build the environment and create a result link as gc-root
+    /// Build the environment with side effects:
+    ///
+    /// - create a result link as gc-root
+    /// - copy catalog.json from the result into the environment
```

Comment:
I think this is the most controversial part of this PR. It makes sense to create the symlink in this method since Nix will do it for us. But that as well as copying the catalog means we're doing a bit more than building.

A related question is whether builds should even be added to the `TemporaryEnvironment`'s `path`. In some sense the symlink is a separate piece of state (`path` represents everything you need to create an environment, and the symlink is a gcroot to the result of having created the environment locally), so we could return `Result<PathBuf, EnvironmentError2>` instead.

I did what seemed simplest, but I wasn't 100% sure what's best so I thought I'd bring it up.

Reply:
@mkenigs and I discussed this offline. I think it's fine to include the catalog and the symlink in the temporary `path`. If you're just doing a build and the stability of a package has moved forward, you would expect that the entry in the lockfile also moves forward (unless you've pinned the package). So in that case you would expect that a build with the same inputs results in a different output. Similarly, if you do a build during an install, obviously it makes sense for the symlink to point to a new store path.

Also, this behavior is easily reversible, so I don't see much risk going ahead with this implementation.

Reply:
Updated the comment to explain